plugin.rb

```diff
@@ -161,13 +161,20 @@ def send_pm(title, text, user)
                         PostDestroyer.new(Discourse.system_user, post).destroy
                     end
                 elsif raw[8, 11] == "code_sample" then
-                    text = "Hello @" + post.user.username + ", please refer to this topic for posting your code sample. Thanks. https://forum.codewizardshq.com/t/how-to-post-code-samples/21423/1"
+                    text = "Hello @" + topic.user.username + ", please refer to this topic for posting your code sample. Thanks. https://forum.codewizardshq.com/t/how-to-post-code-samples/21423/1"
```

Comment:
@sharpkeen where does the `topic` variable come from? Are you sure that exists in this scope?

Reply:
@daniel-schroeder-dev I think I see what you mean. In that case, where would I add `do |topic|`?

Reply:
Not quite. Read the stuff that sets up variables before the conditional logic runs. Do you understand what each variable references? Can you get the topic/username from there?

Reply:
Is it this part that defines it?

`            raw = post.raw
`

Reply:
@daniel-schroeder-dev Also, is the raw[8,13] that I added in my last elsif accurate?

Reply:
> Is it this part that defines it?
> 
> ` raw = post.raw`

Not quite. Keep looking!



> @daniel-schroeder-dev Also, is the raw[8,13] that I added in my last elsif accurate?

That isn't correct. The first number is where the slice should start. The second is how many characters to slice. The slice should always start at position 8, becuase `@system` and a space is 8 characters. The number of characters to slice will be the length of the command string.

Reply:
@daniel-schroeder-dev Hmm, I'm not really sure what defines it. I'm not completely sure if it's the `Discourse.Event.on` parameter. In that case for the raw[], which numbers should I put?

Reply:
> Hmm, I'm not really sure what defines it. 

We need to get the first post in this topic because that'll allow us to get the original poster (OP). If you look at this line, it grabs the first post in this topic: https://github.com/codewizardshq/CWHQ-Discourse-Bot/blob/a8c64fe40967d7668545ec32f7c6cf813ecd2977/plugin.rb#L142.

To get the user information from that `oPost` object, we access the `user` field. You can see that Charlie did that here to see if the topic creator is the same person that asked to close the post: https://github.com/codewizardshq/CWHQ-Discourse-Bot/blob/a8c64fe40967d7668545ec32f7c6cf813ecd2977/plugin.rb#L157-L159

> In that case for the raw[], which numbers should I put?

Here's how the `[]` operator works in Ruby:

```ruby
➜  ~ irb
irb(main):001:0> text = "A line of text"
=> "A line of text"
irb(main):002:0> text[0, 6]
=> "A line"
irb(main):003:0> text[7, 7]
=> "of text"
```

The first number is the index you want to start slicing the string on. The second number is the number of characters you want. The `raw` variable holds the raw text from the post. I don't see where you've used `raw[8, 13]` anywhere in this commit, though.

Reply:
@daniel-schroeder-dev Oh, okay, I see what you mean. In that case, would this work?

```
 elsif raw[8, 11] == "code_sample" then
                    text = "Hello @" + oPost.user.username + ", please refer to this topic for posting your code sample. Thanks. https://forum.codewizardshq.com/t/how-to-post-code-samples/21423/1"
                    create_post(post.topic_id, text)
                    log_command("received code_sample message", "https://forum.codewizardshq.com/t/#{post.topic_id}", oPost.user.username)
                    PostDestroyer.new(Discourse.system_user, post).destroy
                elsif raw[8,12] == "project_link" then
                    text = "Hello @" + oPost.user.username + ", please refer to this topic for posting the link to your project. Thanks. https://forum.codewizardshq.com/t/how-to-post-project-links/21426/1"
                    create_post(post.topic_id, text)
                    log_command("received project_link message", "https://forum.codewizardshq.com/t/#{post.topic_id}", oPost.user.username)
                    PostDestroyer.new(Discourse.system_user, post).destroy
                elsif raw[8,28] == "code_sample_and_project_link" then
                    text = "Hello @" + oPost.user.username + ", please refer to these topics for posting the link to your project and pasting your code. https://forum.codewizardshq.com/t/how-to-post-code-samples/21423/1 and https://forum.codewizardshq.com/t/how-to-post-project-links/21426/1. Thanks."
                    create_post(post.topic_id, text)
                    log_command("received project_link and code_sample message", "https://forum.codewizardshq.com/t/#{post.topic_id}", oPost.user.username)
                    PostDestroyer.new(Discourse.system_user, post).destroy
```

Reply:
Yup, that works!
```ruby
irb(main):001:0> raw = "@system code_sample_and_project_link"
=> "@system code_sample_and_project_link"
irb(main):002:0> raw[8,28]
=> "code_sample_and_project_link"
```

Reply:
Alright, when I get time in a few hours, I'll make a pull request. Thanks!